.github/workflows/rename-module.yml

```
@@ -17,71 +21,96 @@ jobs:
         with:
           fetch-depth: 0 # everything
 
+      - run: git fetch --tags https://github.com/ethereum/go-ethereum.git
+
+      - run: git checkout ${{ inputs.source }}
+
+      - name: References pointing to source
+        # NOTE: This step assumes that the source has been checked out, which
+        # might not hold if reordered.
+        run: |
+          git branch --points-at HEAD;
+          git tag --points-at HEAD;
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: "go.mod"
+
+      - name: Detect Go module
+        id: go
+        run: |
+            echo "MODULE=$(go list -m)" >> "$GITHUB_OUTPUT";
+            echo "MODULE_SUFFIX=$(go list -m | cut -b 12-)" >> "$GITHUB_OUTPUT"; # Strip github.com/
+
+      - name: Validate Go module
+        if: ${{ steps.go.outputs.MODULE != 'github.com/ava-labs/libevm' && steps.go.outputs.MODULE != 'github.com/ethereum/go-ethereum' }}
+        run: echo "Unexpected Go module ${{ steps.go.outputs.MODULE }}" && exit 1;
```

supernit let's exit even if echo fails, no point having `&&`

```suggestion
        run: echo "Unexpected Go module ${{ steps.go.outputs.MODULE }}"; exit 1
```

---

If `echo` were to fail then the step would fail anyway.